packages/dashboard/src/components/tasks/task-cancellation.tsx

```diff
@@ -0,0 +1,93 @@
+import { Button, ButtonProps, Theme, Tooltip } from '@mui/material';
+import { TaskState } from 'api-client';
+import React from 'react';
+import { AppControllerContext } from '../app-contexts';
+import { AppEvents } from '../app-events';
+import { RmfAppContext } from '../rmf-app';
+import { UserProfileContext } from 'rmf-auth';
+import { Enforcer } from '../permissions';
+import { makeStyles, createStyles } from '@mui/styles';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    enableHover: {
+      '&.Mui-disabled': {
+        pointerEvents: 'auto',
+      },
+    },
+  }),
+);
+
+export interface TaskCancelButtonProp extends ButtonProps {
+  taskId: string | null;
+}
+
+export function TaskCancelButton({ taskId, ...otherProps }: TaskCancelButtonProp): JSX.Element {
+  const classes = useStyles();
+  const rmf = React.useContext(RmfAppContext);
+  const appController = React.useContext(AppControllerContext);
+  const profile = React.useContext(UserProfileContext);
+
+  const [taskState, setTaskState] = React.useState<TaskState | null>(null);
+
+  React.useEffect(() => {
+    if (!rmf || !taskId) {
+      return;
+    }
+    const sub = rmf.getTaskStateObs(taskId).subscribe(setTaskState);
+    return () => sub.unsubscribe();
+  }, [rmf, taskId]);
+
+  const taskCancellable =
+    taskState &&
+    taskState.status &&
+    !['canceled', 'killed', 'completed', 'failed'].includes(taskState.status);
+  const userCanCancelTask = profile && Enforcer.canCancelTask(profile);
+
+  function capitalizeFirstLetter(status: string): string {
```

Comment:
Maybe we can move it in the utils.ts file

Reply:
since it's only used here, I'll keep it here for now